src/wp-includes/comment-template.php

```diff
@@ -2031,8 +2039,10 @@ function comment_id_fields( $post_id = 0 ) {
  *                                     being replied to.
  * @param bool         $link_to_parent Optional. Boolean to control making the author's name a link
  *                                     to their comment. Default true.
+ * @param int          $post_id        Optional. The post ID the comment form is being displayed for.
```

Comment:
It'd be good to accept `int|WP_Post` for this too. It's so common developers will assume. Default to the current global.

You can adapt the code in the above funciton.